Modules/IO/MeshVTK/test/itkVTKPolyDataMeshIOGTest.cxx

```diff
@@ -76,18 +76,16 @@ Expect_lossless_writing_and_reading_of_points(const std::string &
   reader->SetMeshIO(itk::VTKPolyDataMeshIO::New());
   reader->Update();
 
-  const auto outputMesh = reader->GetOutput();
-  ASSERT_NE(outputMesh, nullptr);
-  const auto outputPoints = outputMesh->GetPoints();
-  ASSERT_NE(outputPoints, nullptr);
+  const auto & outputMesh = itk::Deref(reader->GetOutput());
```

Comment:
This is a good demonstration of how this is supposed to be used. I would like to see some more instances of use, better demonstrating a need for a new function and exception class.

Reply:
Thanks @dzenanz 

I use `elastix::Deref` extensively within elastix, e.g. [(source)](https://github.com/SuperElastix/elastix/blob/5f0940d2ec5498a6bc98f53c3eaa6a0a1ccd74b3/Common/ImageSamplers/itkImageRandomSampler.hxx#L56-L57):

    elastix::Deref(MersenneTwisterRandomVariateGenerator::GetInstance()).SetSeed(...);

and [(source)](https://github.com/SuperElastix/elastix/blob/5f0940d2ec5498a6bc98f53c3eaa6a0a1ccd74b3/Common/ImageSamplers/itkImageRandomSampler.hxx#L59-L61)

    MultiThreaderBase & multiThreader = elastix::Deref(this->ProcessObject::GetMultiThreader());
    multiThreader.SetSingleMethod(...);
    multiThreader.SingleMethodExecute();

Once `itk::Deref` is available, I would like to replace those `elastix::Deref` calls with `itk::Deref`. I feel that this function isn't really elastix specific. It could be useful for other ITK users as well. Does that answer your question?